core/shared/src/main/scala/zio/logging/internal/FileWriter.scala

```diff
@@ -38,15 +44,50 @@ private[logging] class FileWriter(
   final def write(buffer: Array[Char], offset: Int, length: Int): Unit =
     writer.write(buffer, offset, length)
 
-  final def writeln(line: String): Unit = {
-    writer.write(line)
-    writer.write(System.lineSeparator)
+  private def makeDateFile(): File = {
+    val currentDateTime = LocalDateTime.now()
+    val formatter       = DateTimeFormatter.ofPattern("yyyy-MM-dd")
+    val time            = formatter.format(currentDateTime)
+    val fileNameArray   = destination.getFileName.toString.split("\\.")
+    val timeFileName    = if (fileNameArray.length >= 2) {
+      fileNameArray.dropRight(1).mkString(".") + "-" + time + "." + fileNameArray.last
+    } else {
+      fileNameArray.head + "-" + time
+    }
+    val timeDestination = FileSystems.getDefault.getPath(destination.getParent.toString, timeFileName)
+    timeDestination.toFile
+  }
 
-    entriesWritten += 1
+  final def writeln(line: String): Unit =
+    rolling match {
+      case Some(value) =>
+        value match {
+          case FileRollingPolicy.TimeBasedRollingPolicy =>
+            val output = new OutputStreamWriter(
+              new FileOutputStream(makeDateFile(), true),
+              charset
+            )
```

Comment:
this in general need some major improvement (as I mentioned in my previous comment)
open output stream with each log message (writeln) is expensive  operation ...

also when FileWriter have rolling policy it does not make sense to create non rolling writer (line 34)